### PR TITLE
Set subuid and subgid to a large range for podman rootless smoker con…

### PR DESCRIPTION
**Pull request by ehelms**
…tainers

### CODE REVIEW

File: roles/smoker/tasks/main.yml [hunk 1]
```diff
@@ -4,6 +4,23 @@
...
+  package:
+    name: 'podman'
+
+- name: Set max user namespaces
+  become: true
+  shell: echo 100000 > /proc/sys/user/max_user_namespaces
+
+- name: Set subuid range for user
+  become: true
+  command: usermod --add-subuids 10000-75535 {{ ansible_user }}
```
Comment (evgeni) [inline roles/smoker/tasks/main.yml:18]: it's super confusing that this is a range, but `/etc/subuid` contains `start:length` instead

Comment (ehelms) [reply roles/smoker/tasks/main.yml:18]: That's how the command works (with the `-`)

File: roles/smoker/tasks/main.yml [hunk 2]
```diff
@@ -4,6 +4,23 @@
...
     name: '{{ smoker_browser_packages }}'
 
+- name: 'Install podman'
+  become: true
+  package:
+    name: 'podman'
+
+- name: Set max user namespaces
+  become: true
+  shell: echo 100000 > /proc/sys/user/max_user_namespaces
```
Comment (ekohl) [inline roles/smoker/tasks/main.yml:14]: Isn't this just `user.max_user_namespaces` that can be set via sysctl?
```suggestion
  sysctl:
    name: user.max_user_namespaces
    value: 100000
```

File: roles/smoker/tasks/main.yml (hunk 1 above)
Comment (evgeni) [reply roles/smoker/tasks/main.yml:18]: Yeah, I realize that, just was commenting for whoever will also look at this and wonder why it does not match their memories ;)

File: roles/smoker/tasks/main.yml [hunk 3]
```diff
@@ -4,6 +4,25 @@
...
+  become: true
+  sysctl:
+    name: user.max_user_namespaces
+    value: 100000
+
+- name: Set subuid range for user
+  become: true
+  command: usermod --add-subuids 10000-75535 {{ ansible_user }}
+
+- name: Set subuid range for user
```
Comment (ekohl) [inline roles/smoker/tasks/main.yml:22]: This doesn't match the content
```suggestion
- name: Set subgid range for user
```

**Review by ekohl — COMMENTED** [review]
Can't you combine both usermod commands into a single command?

I also wonder if this should really be a podman role that should be included instead.

**Comment by evgeni** [comment]
I agree that this is probably better suited in an own role. No hard feelings about one or two usermod calls.

File: roles/podman/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  become: true
+  sysctl:
+    name: user.max_user_namespaces
+    value: 100000
+
+- name: Set subuid range for user
+  become: true
+  command: usermod --add-subuids 10000-75535 {{ ansible_user }}
+
+- name: Set subuid range for user
```
Comment (evgeni) [inline roles/podman/tasks/main.yml:16]:
```suggestion
- name: Set subgid range for user
```

**Review by evgeni — APPROVED** [review]

**Review by ekohl — APPROVED** [review]
Feel free to merge when you want to start deploying it.

**Comment by ehelms** [comment]
I'm going to merge it ahead as there will be a follow on to layer in the requirements to run the client tests (e.g. setting the set of clients and building the images)

**Pull request merged**